frappe/query_builder/builder.py

```diff
@@ -0,0 +1,46 @@
+from pypika import MySQLQuery, Order, PostgreSQLQuery, terms
+from pypika.queries import Schema, Table
+
+class common:
+	terms = terms
+	desc = Order.desc
+	Schema = Schema
+
+	@staticmethod
+	def Table(class_name: str, *args, **kwargs) -> Table:
+		if not class_name.startswith("__"):
+			class_name = f"tab{class_name}"
+		return Table(class_name, *args, **kwargs)
+
+
+class MariaDB(common, MySQLQuery):
+	Field = terms.Field
+
+	@classmethod
+	def from_(cls, table, *args, **kwargs):
+		if isinstance(table, str):
+			table = cls.Table(table)
+		return super().from_(table, *args, **kwargs)
+
+
+class Postgres(common, PostgreSQLQuery):
+	field_translation = {"table_name": "relname", "table_rows": "n_tup_ins"}
+	schema_translation = {"tables": "pg_stat_all_tables"}
```

Comment:
Add a comment to better explain the relationship between these variables...and why should exist this way.

maybe add a TODO for scope of simplification? issues we can't resolve